core/unit_test/TestRangePolicyScan.hpp

```diff
@@ -0,0 +1,227 @@
+        check_scan_results();
+        ASSERT_EQ(ValueType(work_size * (work_size - 1) / 2),
+                  return_val);  // sum( 0 .. N-1 )
+      }
+    } else {
+      // Construct RangePolicy for parallel_scan
+      // based on template Args and work_size.
+      Kokkos::RangePolicy<Args...> policy(0, work_size);
+
+      // Input: label, work_count, functor
+      Kokkos::parallel_scan("TestWithStrArg3", policy, *this);
+      check_scan_results();
+
+      // Input: work_count, functor
+      Kokkos::parallel_scan(policy, *this);
+      check_scan_results();
+
+      {
+        // Input: label, work_count, functor
+        // Input/Output: return_value
+        ValueType return_val;
+        Kokkos::parallel_scan("TestWithStrArg4", policy, *this, return_val);
+        check_scan_results();
+        ASSERT_EQ(ValueType(work_size * (work_size - 1) / 2),
+                  return_val);  // sum( 0 .. N-1 )
+      }
+
+      {
+        // Input: work_count, functor
+        // Input/Output: return_value
+        ValueType return_val;
+        Kokkos::parallel_scan(policy, *this, return_val);
+        check_scan_results();
+        ASSERT_EQ(ValueType(work_size * (work_size - 1) / 2),
+                  return_val);  // sum( 0 .. N-1 )
+      }
+
+      // Check Kokkos::Experimental::require()
+      // for one of the signatures.
+      {
+        using Property =
+            Kokkos::Experimental::WorkItemProperty::HintLightWeight_t;
+        const auto policy_with_require =
+            Kokkos::Experimental::require(policy, Property());
+
+        // Input: work_count, functor
+        // Input/Output: return_value
+        ValueType return_val;
+        Kokkos::parallel_scan(policy_with_require, *this, return_val);
+        check_scan_results();
+        ASSERT_EQ(ValueType(work_size * (work_size - 1) / 2),
+                  return_val);  // sum( 0 .. N-1 )
+      }
+    }
+  }
+
+  // Run test_scan() for a collection of work size
+  template <typename... Args>
+  void test_scan(const std::vector<size_t> work_sizes) {
+    for (size_t i = 0; i < work_sizes.size(); ++i) {
+      test_scan<Args...>(work_sizes[i]);
+    }
+  }
+
+  // Operator defining work done in parallel_scan.
+  // Simple scan over [0,1,...,N-1].
+  // Compute both prefix and postfix scans.
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const size_t i, ValueType& update, bool final_pass) const {
+    if (final_pass) {
+      prefix_results(i) = update;
+    }
+    update += i;
+    if (final_pass) {
+      postfix_results(i) = update;
+    }
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  void init(ValueType& update) const { update = 0; }
+
+  KOKKOS_INLINE_FUNCTION
+  void join(ValueType& update, const ValueType& input) const {
+    update += input;
+  }
+};  // struct TestScanRangePolicy
+
+}  // namespace
+
+#ifndef KOKKOS_ENABLE_OPENMPTARGET
+TEST(TEST_CATEGORY, range_policy_scan) {
+  {
+    TestRangePolicyScan<char> f;
```

Comment:
Can we add `TestRangePolicyScan<bool>`.  This would have helped catch a bug in the HIP backend

Reply:
Actually we shouldn't do that. It makes no sense to do a parallel_scan on boolean. I was thinking about parallel_reduce when I wrote that. Sorry for the confusion